js/bbs-api/api_client.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+const messageFunc = (message) => {
+    const elementMessage = document.getElementById("message")
+    elementMessage.innerHTML = 'メッセージ [ ' + message + ' ]'
+}
+
+const getToken = () => {
+    const token = localStorage.getItem("token")
+    const param = 'Bearer ' + token
+    return param
+}
+
+const getFetch = async (host, params) => {
+    await fetch(host, params)
+    .then((response) => {
+        if (!response.ok) {
+            console.error('サーバーエラー')
+        }
+        return response.json()
+    })
+    .then(json => {
+        if (json["massage"]) {
+        messageFunc(json["massage"]) 
+        }
+        if (json["token"]) {
+            localStorage.setItem("token", json["token"]) 
+        }
+    })
+    .catch(error => {
+        console.error('通信に失敗しました', error)
+    })
+}
+
+const registerUser = async (host, name, bio, password) => {
```

Comment:
main.js側の指摘にある'host'系の情報をapi_client.js側に寄せると、引数のhostはすべてメソッド内部に隠せますね

外からパラメータを変更して実行する予定のないものは引数にしないようにします
下記は雰囲気サンプルです

```suggestion
const registerUser = async (name, bio, password) => {
  // hostはファイル全体で定義しておく
   const endPoint = host + "/register";
   ...
```

Reply:
↑↑↑他のところも同じなので確認&修正してみてください